Formula/l/llvm@15.rb

```diff
@@ -27,6 +22,8 @@ class LlvmAT15 < Formula
 
   keg_only :versioned_formula
 
+  deprecate! date: "2024-11-01", because: :versioned_formula
```

Comment:
Probably want to wait a bit for this one -- users on older systems probably need this to get a newer compiler (particularly for users who can't build newer LLVM)

Reply:
Yeah we special case compiler formulae and only really deprecate them when they are long broken (or super low usage).

Reply:
In this case, would like to have a DSL before `deprecate!` to prohibit usage as dependency in core. Something to represent upstream deprecation but not Homebrew deprecated.

Closest thing we have is `livecheck { skip }`. Could potentially make this cause audit failure for new formulae that try to add dependency.

Would be useful for other formulae like `pcre` where we should no longer accept new formulae that use it.

---

> LLVM 14 isn't even deprecated so doesn't really make sense to deprecate 15 first.

I don't think order matters in this case based on our deprecation policy. Depending on actual API changes, it may be better to keep an older version longer than a newer version given `n - 1` may be relatively similar while `n - 2` can be a significant break.

In current state of versioned LLVMs, we actually have 0 non-deprecated usage of odd versions (`llvm@15` and `llvm@17`) while everything non-deprecated is using even ones. Though this really depends on state of formulae that are built on top of specific LLVM (e.g. `ldc`, `pocl`, `rust`).

Reply:
Yea, enforcing no new dependencies without deprecating would be good.